### PR TITLE
[#64] [Chore] Setup DI with Koin

### PR DESCRIPTION
**Pull request by markgravity**
https://github.com/markgravity/kmm-ic/issues/64

## What happened 👀

- Setup DI by using Koin

## Insight 📝

Separated into a small story from https://github.com/markgravity/kmm-ic/issues/15, so it's easier for reviewers to review

## Proof Of Work 📹

CI/CD successfully

### CODE REVIEW

**Review by phongvhd93 — APPROVED** [review]

File: .github/workflows/deploy_production_appstore.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
     steps:
     - uses: actions/checkout@v2
       with:
-        fetch-depth: 0
+        submodules: recursive
```
Comment (suho) [inline .github/workflows/deploy_production_appstore.yml]: Why you changed to `recursive`?

Comment (markgravity) [reply .github/workflows/deploy_production_appstore.yml]: @suho It supposes to fetch our submodules, such as: `jsonapi-kotlin`

Comment (markgravity) [reply .github/workflows/deploy_production_appstore.yml]: @suho reverted it 🙏

File: .github/workflows/deploy_staging_firebase.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ jobs:
     steps:
     - uses: actions/checkout@v2
       with:
-        fetch-depth: 0
```
Comment (suho) [inline .github/workflows/deploy_staging_firebase.yml]: The indent is changed 🤔

File: shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/di/DI.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package co.nimblehq.mark.kmmic.di
+
+import org.koin.core.context.startKoin
+
+fun init(){
```
Comment (suho) [inline shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/di/DI.kt]: Please put attention to details

```suggestion
fun init() {
```

File: .github/workflows/deploy_staging_firebase.yml (hunk 2 above)
Comment (suho) [reply .github/workflows/deploy_staging_firebase.yml]: @markgravity You did not check this

File: shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/di/DI.kt (hunk 3 above)
Comment (suho) [reply shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/di/DI.kt]: @markgravity Same here

File: .github/workflows/deploy_staging_firebase.yml (hunk 2 above)
Comment (markgravity) [reply .github/workflows/deploy_staging_firebase.yml]: @suho The indent is changed to correct the indent 🙏

File: shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/di/DI.kt (hunk 3 above)
Comment (markgravity) [reply shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/di/DI.kt]: I wonder why detekt doesn't show up the warning here 🤔 
Anyway, I updated at [0e58a00](https://github.com/markgravity/kmm-ic/pull/66/commits/0e58a00224649fb8d5266617a4ff0b6b3b3c0212)

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -78,9 +78,14 @@ jobs:
       run: bundle exec fastlane cleanUpOutput
 
     - name: Review pull request by Danger
+      uses: MeilCli/danger-action@v5
+      with:
+        plugins_file: 'ios/Gemfile'
```
Comment (suho) [inline .github/workflows/test.yml:83]: @markgravity Please double check, this has 2 spaces, while the above is 1 space

Comment (markgravity) [reply .github/workflows/test.yml:83]: @suho updated at [c2fc6cd](https://github.com/markgravity/kmm-ic/pull/66/commits/c2fc6cdcfc941b5bb9b4e5e0255fd1d71d0cfc4a)

**Review by AVI5HEK — APPROVED** [review]

**Review by suho — APPROVED** [review]

**Pull request merged**